src/vos/vos_dtx.c

```diff
@@ -412,25 +412,25 @@ dtx_key_rec_exchange(struct umem_instance *umm, struct vos_krec_df *key,
 	umem_tx_add_ptr(umm, tgt_key, sizeof(*tgt_key));
 
 	if (key->kr_bmap & KREC_BF_EVT) {
-		umem_tx_add_ptr(umm, &tgt_key->kr_evt[0],
-				sizeof(tgt_key->kr_evt[0]));
-		tgt_key->kr_evt[0] = key->kr_evt[0];
-
-		umem_tx_add_ptr(umm, &key->kr_evt[0],
-				sizeof(key->kr_evt[0]));
-		memset(&key->kr_evt[0], 0, sizeof(key->kr_evt[0]));
+		tgt_key->kr_evt = key->kr_evt;
+		tgt_key->kr_bmap |= KREC_BF_EVT;
```

Comment:
The KREC_BF_EVT should has already been set when the tgt_key is allocated.

Reply:
changed to assert.   This will all go away when we move to logging punches, right?